src/runtime_src/hip/api/hip_device.cpp

```diff
@@ -66,7 +70,7 @@ hip_init(unsigned int flags)
   std::call_once(device_init_flag, xrt::core::hip::device_init);
 }
 
-static size_t
+static int
```

Comment:
Why `int`.  Should be unsigned and `size_t` is the proper type to use here.
It's possible hipGetDeviceCount is using `int*`, but then you static_cast.

Reply:
Sorry, rebase error. It has been reverted back.